core/src/main/java/com/graphhopper/matrix/algorithm/DijkstraOneToManyMatrixAlgorithm.java

```diff
@@ -0,0 +1,67 @@
+package com.graphhopper.matrix.algorithm;
+
+import com.graphhopper.matrix.DistanceMatrix;
+import com.graphhopper.routing.*;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.TraversalMode;
+import com.graphhopper.routing.util.Weighting;
+import com.graphhopper.storage.Graph;
+
+import java.util.List;
+
+/**
+ * This implementation uses the DijkstraOneToMany algorithm, which can reuse data structures as long
+ * as the same start-node is in use.
+ *
+ * This means we reuse the algorithm/cache per row.
+ * I.e. for each origin, there is a new DijkstraOneToMany / cache created.
+ *
+ * @author Pascal Büttiker
+ */
+public class DijkstraOneToManyMatrixAlgorithm extends AbstractMatrixAlgorithm {
```

Comment:
As said before: I would not want to have this as an option as this explodes for large areas. Instead a simple one-to-many approach via the normal Dijkstra or bidir. Dijkstra class could be implemented.